internal/datastore/postgres/postgres.go

```diff
@@ -547,12 +547,19 @@ func (pgd *pgDatastore) ReadyState(ctx context.Context) (datastore.ReadyState, e
 	}
 
 	if version == headMigration {
+		// Ensure a datastore ID is present. This ensures the tables have not been truncated.
+		uniqueID, err := pgd.datastoreUniqueID(ctx)
+		if err != nil {
+			return datastore.ReadyState{}, fmt.Errorf("database validation failed: %w; if you have previously run `TRUNCATE`, this database is no longer valid and must be remigrated. See: https://spicedb.dev/d/truncate-unsupported", err)
+		}
```

Comment:
truncating tables is really a problem for all datastores. And it's not only the unique ID, the "revision zero" is also needed. Each DB may have its own state written.

This improves the experience with Postgres but does not solve it holistically

Reply:
It is, but we've encountered it primarily with Postgres and so I started there.

And yes, we need both, but if someone is truncating one table its likely the others are as well, so this is more a sanity check

Reply:
I do think this is the right place to solve it (in all of datastores `ReadyState` method). The risk is that folks can do all sorts of unexpected things to the underlying datastore, making assumptions of how it works.

I think this is probably fine for now, but we should open an issue and follow up on all datastores:
- add a test for all datastores
- check that the state of the datastore is in a good place (e.g. there is at least one revision, there exists at least one migration in the migration table, and the stats table is not empty)